### PR TITLE
Add a configurable name to MypyItem node IDs

### LINKED ISSUE #74: Clarify which pytest items are MyPy when running with --verbose flag.
When running with the `--verbose` flag, pytest-flake8 [handily annotates each output line](https://github.com/tholo/pytest-flake8/blob/c485d5e85ac6b2138405378e9763e7025f58b023/pytest_flake8.py#L90) to clarify that it ran that test (i.e. it's not a hand-written test). For example:

```sh
pytest --verbose
```
```
============================= test session starts ==============================
platform linux -- Python 3.8.1, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /usr/local/bin/python
cachedir: .pytest_cache
rootdir: /usr/src/app, inifile: setup.cfg
plugins: flake8-1.0.4, mypy-0.4.2, cov-2.8.1
collecting ... collected 33 items / 2 deselected / 31 selected

Running mypy on 11 files... done with status 0
Success: no issues found in 11 source files

mymodule/__init__::py PASSED                                              [  x%]
mymodule/__init__.py::FLAKE8 PASSED                                       [  x%] <
mymodule/__main__::py PASSED                                              [  x%]
mymodule/__main__.py::FLAKE8 PASSED                                       [ x%] <
…
mymodule/utils.py::module.utils.function_name PASSED      [ x%]
…
mymodule/tests/test_function.py::test_function PASSED [ x%]
…
```

Would it be possible to similarly alter the [`MyPyItem`](https://github.com/dbader/pytest-mypy/blob/e1e4ebd49da513126afd594c96b22c0d321b9369/src/pytest_mypy.py#L84) to similarly mark type-checks as separate from hand-written tests in pytest's verbose output?

### PR DESCRIPTION
**Pull request by dmtucker**
Resolve #74 

I started with #45 to ensure the effort @neozenith put into this work is acknowledged 😊 

The main concern with that PR was primarily the use of the private attribute `_nodeid` and secondarily a forced changed (to absolute paths) of the default node ID. This deals with those by using the `nodeid` kwarg to `pytest.Item` and re-creating a `MypyItem` that will have a name added to the node ID (basing off of the original `MypyItem`'s node ID).

The `nodeid` kwarg is not available until Pytest 3.5, so **this drops support for `pytest >= 2.8, < 3.5`**. That has the convenient side-effect of fixing #46 (again), which was reintroduced by the extra testing in #73.

Lastly, the new node ID name is configurable in the same manner as `mypy_argv`:
``` python
import mypy.version  # beware: not a public API

def pytest_configure(config):
    """Add the mypy version to the node ID of MypyItems."""
    plugin = config.pluginmanager.getplugin('mypy')
    plugin.nodeid_name += mypy.version.__version__
```
```
============================= test session starts ==============================
platform darwin -- Python 3.8.0, pytest-5.3.5, py-1.8.1, pluggy-0.13.1 -- /Users/dmtucker/Projects/pytest-mypy/venv/bin/python
cachedir: .pytest_cache
Using --randomly-seed=1581720682
rootdir: /Users/dmtucker/Projects/pytest-mypy
plugins: xdist-1.29.0, cov-2.5.1, forked-1.1.3, randomly-2.1.1, mypy-0.4.2
collecting ... collected 1 item

conftest.py::mypy0.761 PASSED                                       [100%]
===================================== mypy =====================================

Success: no issues found in 1 source file
============================== 1 passed in 0.20s ===============================
```

Unfortunately, `MypyItem`s (still) cannot be selected by their node ID.
I suspect that is not supported by Pytest (as opposed to something this project is lacking).

### CODE REVIEW

**Comment by dmtucker** [comment]
```
$ pypinfo --days 180 --limit 50 pytest version
Served from cache: False
Data processed: 510.40 GiB
Data billed: 510.40 GiB
Estimated cost: $2.50
```
| version | download_count |
| ------- | -------------- |
| 5.3.2   |      6,363,694 |
| 5.1.2   |      4,644,754 |
| 5.2.1   |      4,355,028 |
| 5.2.2   |      3,837,703 |
| 4.4.1   |      3,685,175 |
| 5.3.1   |      3,330,827 |
| 4.6.3   |      2,866,086 |
| 4.6.5   |      2,807,463 |
| 5.3.5   |      2,695,021 |
| 3.6.3   |      2,465,968 |
| 5.0.1   |      2,368,811 |
| 5.2.0   |      2,335,403 |
| 5.1.1   |      2,331,197 |
| 4.6.6   |      2,122,345 |
| 5.3.4   |      1,937,956 |
| 4.6.9   |      1,608,039 |
| 5.1.3   |      1,594,876 |
| 5.3.0   |      1,325,294 |
| 3.8.2   |      1,240,005 |
| 3.10.1  |      1,200,953 |
| 4.5.0   |      1,016,453 |
| 3.5.0   |        828,985 |
| 3.6.4   |        781,930 |
| 4.4.0   |        747,561 |
| 4.0.1   |        746,050 |
| 4.0.2   |        691,049 |
| 5.2.4   |        681,818 |
| 4.3.0   |        662,560 |
| 4.6.7   |        633,568 |
| 3.2.3   |        600,155 |
| 4.6.8   |        592,262 |
| 4.3.1   |        579,808 |
| 5.3.3   |        560,702 |
| 3.8.1   |        529,150 |
| 5.1.0   |        525,563 |
| 3.5.1   |        438,120 |
| 3.0.7   |        392,946 |
| 4.1.1   |        350,118 |
| 4.6.4   |        345,728 |
| 3.6.2   |        328,139 |
| 4.4.2   |        315,748 |
| 4.6.2   |        297,390 |
| 3.4.0   |        283,365 |
| 4.0.0   |        276,685 |
| 3.1.1   |        261,327 |
| 3.9.1   |        255,702 |
| 3.3.1   |        252,051 |
| 2.9.1   |        244,512 |
| 3.8.0   |        241,802 |
| 5.0.0   |        232,492 |
| Total   |     69,810,337 |

From that list, we wouldn't support these:

| version | download_count |
| ------- | -------------- |
| 3.2.3   |        600,155 |
| 3.0.7   |        392,946 |
| 3.4.0   |        283,365 |
| 3.1.1   |        261,327 |
| 3.3.1   |        252,051 |
| 2.9.1   |        244,512 |
| Total   |      2,034,356 |

In other words, we'd still support ~97% of the Pytest downloads from the last 6mos.
